### ISSUE TITLE
Survey Completion Percentage Display on the Dashboard

### ISSUE THREAD

**Issue by ertjlane** [BIDMCDigitalPsychiatry/LAMP-platform#777]
For each activity, in the same section in which admin/investigator users may assign a notification schedule, they will also have the option to select an activity completion rate and timeframe. I.E. a box allowing them to select the number of times the activity should be completed, and directly adjacent to that box, the timeframe of completion. For example, a daily survey to be completed 7 times, in 1 week. This proportion is then used to calculate the percentage completed, which will display to the user in the activity completion graph shown in the portal section. 

Steps previously discussed to accomplish this and timeline are listed below:
Add/edit percentage settings in the admin section - UI & development - 2.5 days.
API changes to get the percentage based on generated activity event submissions in a particular interval of time - 2 days.
Showing percentage value in patient dashboard - 0.5 day
Developer testing and fixes - 1 day
QA and release – 2 days

Prior Mockup Screenshots posted here for reference. Revisions still needed.
![Lamp (1)](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/assets/104088715/011bdec5-5bc2-4fd8-a0fe-8cffef08a4a1)

![Lamp2](https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/assets/104088715/e8e7b9dd-f28b-4c50-b257-e0512b0749a6)

In addition, it would be apt to review a new mockup of the UI, incorporating the changes discussed to make sure that we are in agreement about the final output!

**Comment by avaidyam** [comment]
I understand the UI changes but @ertjlane can you more thoroughly detail what you mean by API changes?

**Comment by sarithapillai8** [comment]
@avaidyam We were planning to have a new API for calculating the percentage value. Do you want to handle those in the front end itself?

**Comment by avaidyam** [comment]
@sarithapillai8 Yes, API changes are forbidden in the LAMP Platform. Please reuse existing tools, for example, by adding a Tag to the Activity object with completion requirements, and then check ActivityEvent count vs. that.

**Comment by sarithapillai8** [comment]
@ertjlane can we show the exact percentage value that can be above 100, if the user submits a survey more than what was set in the required field?

**Comment by sarithapillai8** [comment]
@ertjlane FYI We have done like the percentage calculation will start from the time we save the settings for percentage. The duration(days/months/hours/weeks) will be calculated from the time we save these settings.

**Comment by evinkalengaden** [comment]
Hi,
 We have updated in dashboard staging, Pls review and let us know.

Thanks,
Evin K Alengaden

**Comment by ertjlane** [comment]
Hello, thanks all for your patience. I have just reviewed the survey completion activity and have a few questions. I noticed once instance in which a participant had completed an activity within the past 5 days assigned to be completed once per week, but the dashboard demonstrated 0% completion.
<img width="776" alt="Screen Shot 2023-10-26 at 2 00 34 PM" src="https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/assets/104088715/b2e76e0f-d69c-423f-ae44-b737e69f5f88">
<img width="405" alt="Screen Shot 2023-10-26 at 2 00 17 PM" src="https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/assets/104088715/fd23b9ba-6f1a-453b-a2be-78df24ab15f0">


However, there are other instances in which a participant had completed the same activity, with the same 1 week timeframe, once in the past two days and had a 100% completed mark on the dashboard, as they should have. Similarly, a user assigned to a daily survey once/day has a 100% completion mark, despite having last completed the survey 2 days ago. 

<img width="773" alt="Screen Shot 2023-10-26 at 1 58 34 PM" src="https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/assets/104088715/5076cca4-c0c9-4441-8f73-f67fcd8025c3">
<img width="404" alt="Screen Shot 2023-10-26 at 1 59 57 PM" src="https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/assets/104088715/f8f190e5-2836-47fa-8c2d-c40aca8f2980">

**Comment by sarithapillai8** [comment]
@ertjlane Just for clarification, did you set the percentage settings yesterday? The percentage calculation starts from the time we save the settings. I have to check if there is any issue with the timezone. I shall check and let you know.

**Comment by sarithapillai8** [comment]
@ertjlane could you please check the above comment?

**Comment by ertjlane** [comment]
@sarithapillai8 Yes, I set the percentage completion the day I assessed it, I don't think that explains the user with the daily survey who hadn't completed in the last day, but had 100% completion, as even if I assigned it that day they wouldn't have completed any. 
Thank you for investigatin the time zone.

**Comment by ZCOEngineer** [comment]
We didnt get this issue when we checked, QA is testing more on this.

**Comment by sarithapillai8** [comment]
@ertjlane 

> a user assigned to a daily survey once/day has a 100% completion mark, despite having last completed the survey 2 days ago.

A survey can have multiple schedules, the percentage settings will be considered independent of schedule type as it is common to all schedules. There can be a weekly, or daily schedule for a single survey. What we understood was we would be giving a completion count that is to be completed in a particular timeframe.

ie, if the count is given as 2 and the timeframe is one day, within one day the survey needs to be completed 2 times. 
 If the count is 2 and the timeframe is 1 week, then within a week, 2 times the survey needs to be completed. 

This calculation will be started from the time we save the percentage settings, which was already commented above https://github.com/BIDMCDigitalPsychiatry/LAMP-platform/issues/777#issuecomment-1766417179

**Comment by sarithapillai8** [comment]
@ertjlane We got the issue and working on the fix. We can update this by early next week.

**Comment by evinkalengaden** [comment]
@ertjlane We have updated fix in the dashboard staging, Could you please check and let us know.

Thankyou

**Comment by ertjlane** [comment]
Hello all, due to external discussions about the utility of this feature in the context of the study for which it was requested I'm going to put this on hold (hence the delay with my review).

**Comment by avaidyam** [comment]
Alternatively it should be commented out of the codebase with information on how to re-enable it.

**Comment by sarithapillai8** [comment]
@ertjlane We have commented  the section in the code base and updated dashboard staging.

**Comment by ertjlane** [comment]
@sarithapillai8 Perfect thank you!
